KTB-09-refactoring

DenisGubernatorov opened the pull request:
Разнесение логики работы тренажера и вывода сообщений в консоль 1) Добавлен класс LearnWordTrainer с методами отвечающими за логику формирования статистки и вопросов 2) Добавлен дата класс Statistic отвечающий за хранение статистики во время клиентской сессии 3) Добавлен класс Question, отвечающий за создание вопроса

DenisGubernatorov committed:
KTB-09-refactoring
Разнесение логики работы тренажера и вывода сообщений в консоль
1) Добавлен класс LearnWordTrainer с методами отвечающими за логику формирования статистки и вопросов
2) Добавлен дата класс Statistic отвечающий за хранение статистики во время клиентской сессии
3) Добавлен класс Question, отвечающий за создание вопроса

DenisGubernatorov committed:
KTB-09-refactoring
1) Исправление работы функции-расширения   asConsoleString для Question

kaznikit commented on src/main/kotlin/LearnWordTrainer.kt:
```diff
@@ -0,0 +1,91 @@
+data class Question(
+    val variants: List<Word>,
+    val correctAnswer: Word,
+    val isEmptyQuestion: Boolean = false,
```
а для чего это поле если оно нигде не используется? давай удалим

kaznikit commented on src/main/kotlin/LearnWordTrainer.kt:
```diff
@@ -0,0 +1,91 @@
+)
+
+class LearnWordTrainer {
+    private val wordsFile = File("words.txt")
```
давай название файла, количество правильных ответов для определения выучено ли слово (3) и вариантов ответа (4) вынесем в аргументы конструктора и дадим им дефолтные значения

kaznikit commented on src/main/kotlin/LearnWordTrainer.kt:
```diff
@@ -0,0 +1,91 @@
+class LearnWordTrainer {
+    private val wordsFile = File("words.txt")
+    private val dictionary = loadDictionary()
+    private var wordsToLearn = updateWordsToLearn()
```
мне кажется этот список лучше получать заново в каждой из функций, так как он не будет изменяться после выполнения какой-либо из функций и есть вероятность что он будет устаревший

DenisGubernatorov committed:
KTB-09-refactoring
Доработка кода согласно комментариям
1) В Question удалено неиспользуемое свойство
2) Файл, количество правильных ответов, количество вариантов сделаны как аргументы конструктора с дефолтными значениями
3) Список слов для изучения обновляется при получении списка вопросов

DenisGubernatorov commented:
В функции getAnswerVariants не стал обновлять список wordsToLearn, так его уже обновил в getQuestions
